07_text_editors/1_setup.md

```diff
@@ -0,0 +1,81 @@
+# Get started with Python in VS Code (0h20)
+
+The first thing you need after installing VS Code is to install the [Python extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python). This extension is the official Python plugin for VS Code and it provides a lot of useful features.
+
+The second thing you need to do is to select a Python interpreter. 9 out of 10 times your get a `ModuleNotFound` error is because you are using the wrong interpreter. The Python extension will try to detect the interpreter for you, but you can also select it manually.
+
+Make sure you read about deal with environments and the Python interpreter in VS Code: [VSCode docs: Environments](https://code.visualstudio.com/docs/python/environments)
+
+## Setting up VS Code for Python development (0h10)
+
+### Configuration
+
+VSCode is highly customizable. This customization lives in the `.vscode/settings.json` file and can be altered either in that file directly or via the user interface.
+
+> Make sure you have added the `.vscode/` directory to the `.gitignore` file, so your settings don't end up polluting the Git repository by mistake.
+
+Here are some useful settings:
```

Comment:
That would be a matter of personal preference then. I'm not really convinced.

Btw, this would be a good use case for prefixing the code review commend with a "**suggestion** prefix. However, if this is a request change, it's useful for you to mention your reasoning.

My reasoning for keeping the settings first is that I find this styles more engaging. 

First you expose the student to something they don't know. They might get frustrated by hopefully their will try to make some sense out of it before moving on. Then then read the explanation. At this time, I expect the a-ha moment to happen and so they return to read the example again. So I'm doubling their exposure time to new content at the same time I set their expectations on how to deal with unknowns when they find concepts they aren't familiar with.

Reply:
Yes sorry, I meant it as a suggestion of course hehe. 

**nitpick** - My reasoning for this is that most of the content elsewhere is formatted differently so I found myself reading descriptions to then go to the next piece of code and realise the description I read belonged to the previous one. This is because texts are normally read from up-down in the english and portuguese languages. 

As a personal preference, I am more inclined to make the learning easier not to discourage students, I find this "out-of-context-content" annoying and frustrating and it makes it even more confusing for me. But this is what I feel when I read this format, maybe other people will love it. What happened was that in the end I ended up skipping the code, reading the description first and then reading the code for the first time.

**suggestion** - In case you want to maintain this strategy, I would suggest to separate each (code + explanation) block from each other to stop the reader from continuing. I would find that useful in dealing with such structure.

Reply:
Like so?